x-pack/plugins/upgrade_assistant/public/components/tabs/checkup/deprecations/reindex/flyout/warnings_step.tsx

```diff
@@ -107,6 +107,35 @@ export class WarningsFlyoutStep extends React.Component<
 
           <EuiSpacer />
 
+          {warnings.includes(ReindexWarning.apmReindex) && (
+            <EuiText>
+              <EuiCheckbox
+                id={idForWarning(ReindexWarning.apmReindex)}
+                label={<strong>This index will be converted to ECS format</strong>}
+                checked={checkedIds[idForWarning(ReindexWarning.apmReindex)]}
+                onChange={this.onChange}
+              />
+              <p className="upgWarningsStep__warningDescription">
+                Starting in version 7.0.0, APM data will be represented in the Elastic Common
+                Schema. In order for legacy data to be included, it is required that the data is
+                re-indexed to support this new format.
```

Comment:
I think we should probably say something here like "historical APM data is not visible until it is reindexed"